src/manifest/representation.ts

```diff
@@ -111,6 +111,14 @@ class Representation {
    */
   public codec : string | undefined;
 
+  /**
+   * Supplemental codecs are defined as backwards-compatible codecs enhancing
+   * the experience of a base layer codec
+   */
+  public supplementalCodec?: string | undefined;
+
+
+  public isSupplementalCodecSupported: boolean;
```

Comment:
We may want to add a comment to explain what it is for.

If you're not inspired, the HLS spec has a good description https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-4.4.6.2 for the similarly named `SUPPLEMENTAL-PROPERTY` attribute.